Mathlib/Order/Zorn.lean

```diff
@@ -95,98 +98,67 @@ section Preorder
 
 variable [Preorder α]
 
-theorem zorn_preorder (h : ∀ c : Set α, IsChain (· ≤ ·) c → BddAbove c) :
-    ∃ m : α, ∀ a, m ≤ a → a ≤ m :=
+theorem zorn_le (h : ∀ c : Set α, IsChain (· ≤ ·) c → BddAbove c) : ∃ m : α, IsMax m :=
   exists_maximal_of_chains_bounded h le_trans
 
-theorem zorn_nonempty_preorder [Nonempty α]
-    (h : ∀ c : Set α, IsChain (· ≤ ·) c → c.Nonempty → BddAbove c) : ∃ m : α, ∀ a, m ≤ a → a ≤ m :=
+theorem zorn_le_nonempty [Nonempty α]
+    (h : ∀ c : Set α, IsChain (· ≤ ·) c → c.Nonempty → BddAbove c) : ∃ m : α, IsMax m :=
```

Comment:
I can see the appeal of leaving it in `IsMax` form to access API, but it felt like most of the time this condition was deconstructed in the proof anyway. Should we at least have both versions?

Reply:
It's more the abstraction unifying `PartialOrder` and `Preorder` that I thought was an improvement. Does two versions mean three versions?

Reply:
Are there places where you are actually making use of `IsMax` API? Or are you always destructing it? If the latter, I suggest switching it back to the forall statement because it means it's just getting in the way.

Reply:
My point is that there are two possible forall statements - the one with `le -> le` for preorders and the other with `le -> eq` for partial orders. The first is more general but less convenient and the second vice versa. Using `IsMax` abstracts away the problem, and stops us having to pick one of the two in theorem statements.

Reply:
Oh, I completely misunderstood! Thanks for the explanation. This is great as is then!